File: Day1/challenge1/main.go
package main

import (
	"fmt"
	"os"
	"strings"
)

// file path: Day1/challenge1/input.txt

func main() {
	var sum int = 0

	for _, line := range readInputFile("Day1/challenge1/input.txt") {
		var digits []int = []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}

		// get first number in line
		var firstNumber int = 0
		for i := 0; i < len(line); i++ {
			if contains(digits, int(line[i]-'0')) {
				firstNumber = int(line[i] - '0')

				break
			}
		}

		// get second number in line
		var secondNumber int = 0
		for j := len(line) - 1; j >= 0; j-- {
			if contains(digits, int(line[j]-'0')) {
				secondNumber = int(line[j] - '0')
				break
			}

		}

		sum += (firstNumber * 10) + secondNumber

	}
	fmt.Println("Sum of all numbers in file is: ", sum)
}
func readInputFile(fileName string) []string {
	data, err := os.ReadFile(fileName)
	if err != nil {
		fmt.Println("File reading error", err)
		return nil
	}

	// Convert the input data to a string
	input := string(data)

	// Split the input string into lines
	lines := strings.Split(input, "\n")

	return lines
}

func contains(slice []int, val int) bool {
	for _, item := range slice {
		if item == val {
			return true
		}
	}
	return false
}

File: Day1/challenge1/main.py
import re
import time


def readInputFile(fileName):
    with open(fileName) as f:
        lines = f.readlines()
    return lines


def main():
    sum = 0

    start_time = time.time()

    for line in readInputFile("Day1/challenge1/input.txt"):
        digits = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
        for char in line:
            if char in digits:
                firstNumber = char
                break

        for char in reversed(line):
            if char in digits:
                secondNumber = char
                break
        sum += (int(firstNumber) * 10) + int(secondNumber)

    end_time = time.time()

    print("Sum of all numbers in file is:", sum)
    print("Execution time:", (end_time - start_time) * 1000, "milliseconds")


main()
